jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -640,10 +642,8 @@ def apply_RGB_presets(self):
         if n_visible > 2:
             default_opacity = 1 / math.log2(n_visible)
         # Sample along a colormap if we have too many layers
-        if n_visible == 2:
-            preset_colors = [preset_colors[1], preset_colors[4]]
-        elif n_visible == 3:
-            preset_colors = [preset_colors[1], preset_colors[2], preset_colors[4]]
+        if n_visible >= 2 and n_visible < len(preset_colors):
+            preset_colors = [preset_colors[i] for i in preset_inds[n_visible]]
         elif n_visible > len(preset_colors):
             cmap = matplotlib.colormaps['gist_rainbow'].resampled(n_visible)
             preset_colors = [matplotlib.colors.to_hex(cmap(i), keep_alpha=True) for

```

Comment:
this might now read better and be harder to break in the opposite order (`if n_visible > len(preset_color): ... elif n_visible > 2: ...`)

Reply:
ah, I wasn't thinking about the `==` case, so doesn't really make a difference I guess 🤷‍♂️

Reply:
Well, I'm not changing it back!